crates/starknet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -184,11 +184,9 @@ impl JsonRpcHandler {
         match self.api.starknet.read().await.get_class(block_id.into(), class_hash) {
             Ok(contract_class) => Ok(contract_class.try_into()?),
             Err(Error::NoBlock) => Err(ApiError::BlockNotFound),
-            Err(
-                Error::ContractNotFound
-                | Error::StateError(StateError::NoneContractState(_))
-                | Error::NoStateAtBlock { block_number: _ },
-            ) => Err(ApiError::ContractNotFound),
+            Err(Error::StateError(_) | Error::NoStateAtBlock { block_number: _ }) => {
```

Comment:
If I understand correctly, this fixes the problem of some `StateError`s being caught by the `Err(unknown_error)` arm?

Reply:
yes, in the case of get_class it returns 2 variants of Error::StateError.